### PR TITLE
test: improve testing

### PR DESCRIPTION
**Pull request by rdumont**
Two aspects were addressed here:
1. Static type analysis prevents issues such as https://github.com/escaletech/circleci-gh-conventional-release/pull/5#discussion_r451676109. This is most important for the `bin/createRelease.js` file, which stitches parts of the software together.
2. Cover edge cases of creating a release with unit tests

### CODE REVIEW

File: lib/release.js [hunk 1]
```diff
@@ -1,57 +1,49 @@
...
 
-  return gh.createRelease(targetTag, body)
+    default:
+      return { id: 'other', heading: ':alien: Other', weight: 0 }
+  }
 }
+
+module.exports.defaultFormatter = defaultFormatter
+module.exports.defaultCategorizer = defaultCategorizer
+module.exports.composeReleaseBody = composeReleaseBody
```
Comment (weslleyss12) [inline lib/release.js:49]: `NIT` You also can do something like this
```
module.exports = {
  defaultFormatter,
  defaultCategorizer,
  composeReleaseBody
}
```

File: bin/createRelease.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+`.trim()
+
+async function main () {
+  const config = getConfig()
+
+  try {
+    const gh = new GitHubService(
+      new Octokit({ auth: config.gitHubToken }),
+      config.repoOwner,
+      config.repoName)
```
Comment (weslleyss12) [inline bin/createRelease.js:22]: `NIT` I think we could add some lint to this project in the future.

Comment (rdumont) [reply bin/createRelease.js:22]: It does use standard. But this line is awkward indeed.
